### PR TITLE
Add support for default value and enum validation

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6861
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6862
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6863

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [x] Checked native-image compatibility

### CODE REVIEW

File: ballerina/data_item.bal [hunk 1]
```diff
@@ -59,6 +59,14 @@ isolated distinct class DataItem {
         'class: "io.ballerina.lib.copybook.runtime.converter.Utils"
     } external;
 
+    isolated function getPossibleEnumValues() returns string[]? = @java:Method {
+        'class: "io.ballerina.lib.copybook.runtime.converter.Utils"
+    } external;
+
+    isolated function getDefaulValue() returns string? = @java:Method {
```
Comment (DimuthuMadushan) [inline ballerina/data_item.bal:66]:
```suggestion
    isolated function getDefaultValue() returns string? = @java:Method {
```

File: ballerina/default_value_creator.bal [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        string groupItemDefaultValue = string:'join("", ...self.defaultValueFragments);
+        defaultValues.push(self.generateRepeatedString(groupItemDefaultValue, groupItem.getElementCount()));
+        self.defaultValueFragments = defaultValues;
+    }
+
+    isolated function visitDataItem(DataItem dataItem, anydata data = ()) {
+        if dataItem.getRedefinedItemName() is string {
+            return;
+        }
+        string dataItemDefaultValue = dataItem.getDefaulValue() ?: "";
```
Comment (DimuthuMadushan) [inline ballerina/default_value_creator.bal:42]:
```suggestion
        string dataItemDefaultValue = dataItem.getDefaulValue() ? : "";
```

File: ballerina/tests/resources/copybook-json/invalid-copybook-11.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "REQUIRED": "T"
+}
```
Comment (DimuthuMadushan) [inline ballerina/tests/resources/copybook-json/invalid-copybook-11.json:3]:
```suggestion
}

```

File: native/src/main/java/io/ballerina/lib/copybook/runtime/converter/Utils.java [hunk 4]
```diff
@@ -175,6 +180,24 @@ public static BMap<BString, Object> getRedefinedItems(BObject bObject) {
...
+    public static Object getPossibleEnumValues(BObject bObject) {
+        DataItem dataItem = (DataItem) bObject.getNativeData(NATIVE_VALUE);
+        if (dataItem.isEnum()) {
+            List<String> values = dataItem.getPossibleEnumValues();
+            return StringUtils.fromStringArray(values.toArray(new String[0]));
+        }
+        return null;
+    }
+
+    public static Object getDefaulValue(BObject bObject) {
```
Comment (DimuthuMadushan) [inline native/src/main/java/io/ballerina/lib/copybook/runtime/converter/Utils.java:192]:
```suggestion
    public static Object getDefaultValue(BObject bObject) {
```

File: commons/src/main/java/io/ballerina/lib/copybook/commons/schema/SchemaBuilder.java [hunk 5]
```diff
@@ -151,10 +168,15 @@ public CopybookNode visitDataDescriptionEntryFormat1(DataDescriptionEntryFormat1
             return new GroupItem(level, name, occurs, redefinedItemName, getParent(level));
         }
         PictureStringContext pictureType = pictureClause.pictureString();
-        validatePicture(pictureType);
-        return new DataItem(level, name, Utils.getPictureString(pictureType), Utils.isNumeric(pictureType),
-                            Utils.getReadLength(pictureType), occurs, Utils.getFloatingPointLength(pictureType),
-                            redefinedItemName, getParent(level));
+                validatePicture(pictureType);
+        int readLength = Utils.getReadLength(pictureType);
```
Comment (DimuthuMadushan) [inline commons/src/main/java/io/ballerina/lib/copybook/commons/schema/SchemaBuilder.java:172]:
```suggestion
        validatePicture(pictureType);
        int readLength = Utils.getReadLength(pictureType);
```

File: commons/src/main/java/io/ballerina/lib/copybook/commons/schema/SchemaBuilder.java [hunk 6]
```diff
@@ -75,7 +79,10 @@ class SchemaBuilder implements CopybookVisitor<CopybookNode> {
     private GroupItem possibleParent;
     private final Set<String> redefinedItemNames = new HashSet<>();
     private final List<String> errors = new ArrayList<>();
-
+    private DataItem possibleEnum;
+    private int rootLevel = -1;
+    private static final String SPACE = " ";
+    private static final String ZERO = "0";
```
Comment (DimuthuMadushan) [inline commons/src/main/java/io/ballerina/lib/copybook/commons/schema/SchemaBuilder.java:85]:
```suggestion
    private static final String ZERO = "0";

```

File: ballerina/default_value_creator.bal (hunk 2 above)
Comment (MohamedSabthar) [reply ballerina/default_value_creator.bal:42]: ?: called elvis operator no need to add space in between.

**Review by DimuthuMadushan — DISMISSED** [review]

File: commons/src/main/java/io/ballerina/lib/copybook/commons/schema/SchemaBuilder.java (hunk 6 above)
Comment (MohamedSabthar) [reply commons/src/main/java/io/ballerina/lib/copybook/commons/schema/SchemaBuilder.java:85]: Fixed [99012f8](https://github.com/ballerina-platform/module-ballerinax-copybook/pull/39/commits/99012f8e404e40cc605f85618847eef7446ec1a8)

**Review by DimuthuMadushan — APPROVED** [review]

**Pull request merged**